pkg/envfuncs/resource_funcs.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envfuncs
+
+import (
+	"context"
+
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+// SetupCRDs is provided as a helper env.Func handler that can be used to setup the CRDs that are required
+// to process your controller code for testing. This env.Func helper also provides support for
+// injecting resources.CreateOption
+func SetupCRDs(crdPath, pattern string, opts []resources.CreateOption, decodeOpts ...decoder.DecodeOption) env.Func {
```

Comment:
I would keep `SetupCRDs` simple and remove the options, like:

```go
SetupCRDs(crdPath, pattern string) env.Func{...}
```
If someone needs those options, the person can use `decoder.ApplyWithManifestDir` directly to take control.

Reply:
All right. Changes published as suggested